File: binarySearch.js
const binarySearch = (numbers, target) => {
    let low = 0
    let high = numbers.length - 1
    while (low <= high) {
           const middle = Math.floor((low + high) / 2)
           if (target < numbers[middle]) {
            high = middle -1
           } else if (target > numbers[middle]) {
            low = middle + 1
           } else {
            return middle
           }
  };
  return -1
}

console.log(binarySearch([-1,0,3,5,9,12], 9))

File: Python LeetCode/binarySearch.py
def binarySearch(array, target):
    left = 0
    right = len(array) - 1

    # middle = len(array) // 2
    # print(middle)
    while left <= right:
        middle = (left + right) // 2
        if array[middle] == target:
            return middle
        elif target < array[middle]:
            right = middle - 1
        else:
            left = middle + 1
    return middle


array = [0, 1, 21, 33, 45, 45, 61, 71, 72, 73]

target = 33

print(binarySearch(array, target))
